firmware/Core/Src/gps/gps_internal_drivers.c

```diff
@@ -0,0 +1,144 @@
+#include "main.h"
+
+#include "debug_tools/debug_uart.h"
+#include "debug_tools/debug_i2c.h"
+#include "gps/gps_internal_drivers.h"
+#include "uart_handler/uart_handler.h"
+#include "stm_drivers/timing_helpers.h"
+#include "uart_handler/uart_handler.h"
+#include "log/log.h"
+
+#include <stdint.h>
+#include <string.h>
+
+extern UART_HandleTypeDef *UART_gps_port_handle;
+
+const uint32_t GPS_RX_TIMEOUT_MS = 800;
+
+/// @brief Sends a log command to the GPS, and receives the response.
+/// @param cmd_buf log command string to send to the GPS.
+/// @param cmd_buf_len Exact length of the log command string.
+/// @param rx_buf Buffer to store the response.
+/// @param rx_buf_len Length of the response buffer.
+/// @param rx_buf_max_size Maximum length of the response buffer.
+/// @return 0 on success, >0 if error.
+/// @note This function is intended for "once" log commands
+uint8_t GPS_send_cmd_get_response(const char *cmd_buf, uint8_t cmd_buf_len, uint8_t rx_buf[], uint16_t rx_buf_len, const uint16_t rx_buf_max_size)
+{
+	
+	// Reset the GPS UART interrupt variables
+	GPS_set_uart_interrupt_state(0); // Lock writing to the UART_gps_buffer while we memset it
+	for (uint16_t i = 0; i < UART_gps_buffer_len; i++)
+	{
+		// Clear the buffer
+		// Can't use memset because UART_gps_buffer is volatile
+		UART_gps_buffer[i] = 0;
+	}
+	UART_gps_buffer_write_idx = 0;		 // Make it start writing from the start
+	GPS_set_uart_interrupt_state(1); // We are now expecting a response
```

Comment:
Proposal: Move this to AFTER the HAL_UART_Transmit; that way, if the transmit has a failure return code, we don't need to worry about deactivating this interrupt.

Reply:
This makes sense. Done